### DISCUSSION TITLE
How to Fix Picture Problems in 4.16 on macOS Sonoma?

### DISCUSSION THREAD

**Discussion by babyluoyi** [eclipse-platform/eclipse.platform.swt#1170] [Q&A] [answered]
splash screen:
<img width="800" alt="sashulinDown" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/23548657/91089d18-9d64-45cb-8115-70d23e2e9f1c">

**Comment by merks** [comment, marked as answer]
I’m pretty sure this is fixed in the latest release 4.31, isn’t it?

**Reply by sravanlakkimsetti** [reply]
Thats correct @merks it is fixed in 4.31.

**Reply by Phillipus** [reply]
Not "fixed" but worked around. The [underlying cause](https://github.com/eclipse-platform/eclipse.platform.swt/issues/772) remains.

**Comment by babyluoyi** [comment]
@merks thanks.
A lot of exceptions were reported in 4.31.
`	public static Image getImage(String fileName) {
		Bundle bundle = Platform.getBundle(Activator.PLUGIN_ID);
		URL fileURL = bundle.getEntry(fileName);
		if (fileURL == null) return null;
        ImageDescriptor imaged = ImageDescriptor.createFromURL(fileURL);
        return imaged.createImage();
	}`

java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:771)
	at org.eclipse.jface.resource.URLImageDescriptor.createImage(URLImageDescriptor.java:300)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:289)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:267)

**Reply by merks** [reply]
Problems are only fixed in master.  If there are problems in 4.31 they should be reported so they can be fixed for the 4.32 release.  Perhaps this problem is already fixed and is available in a 4.32 integration build.
